### PR TITLE
parameterize log entries

### PR DESCRIPTION
**Pull request by Phil-Friderici**
fix issue https://github.com/ghoneycutt/puppet-module-rsyslog/issues/63

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,13 @@ Path to pid file.
 
 - *Default*: based on platform
 
+log_entries
+-----------
+Array of log entry lines for the rules section. One entry per line. It would be wise to keep it in sync with $logrotate_syslog_files option.
+'USE_DEFAULT' provides the module previous defaults values.
```
Comment (ghoneycutt) [inline README.md]: USE_DEFAULTS - missing 'S'

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -293,6 +293,43 @@
...
+            :kernel            => 'Linux',
+            :osfamily          => 'Debian',
+            :lsbmajdistrelease => '7',
+          }
+        end
+
+        it do
+          expect {
+            should contain_class('rsyslog')
+          }.to raise_error(Puppet::Error)
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: Should include the error message. You can find this by putting `Puppet::Error,/asdf/)`, running the spec tests and seeing what the error is.

File: templates/rsyslog.conf.erb [hunk 3]
```diff
@@ -55,23 +55,9 @@ $ActionSendStreamDriverPermittedPeer <%= @permitted_peer %>
...
-authpriv.*                                              /var/log/secure
-
-# Log all the mail messages in one place.
-mail.*                                                  -/var/log/maillog
-
-# Log cron stuff
-cron.*                                                  /var/log/cron
+<% @log_entries_real.each do |log_entry| -%>
+<%= log_entry %>
+<% end -%>
```
Comment (ghoneycutt) [inline templates/rsyslog.conf.erb]: This should include to the line about `local7.*` The class will need separate default settings for versions < 5 and >= 5 to take care of the `*.emerg` going to `*` or `:omusrmsg:*`

Comment (Phil-Friderici) [inline templates/rsyslog.conf.erb]: On the costs of a sequence change, I have moved just the 'passive' lines. Hope that's ok.

File: manifests/init.pp [hunk 4]
```diff
@@ -222,6 +223,38 @@
   }
   validate_absolute_path($pid_file_real)
 
+  $default_log_entries = [
+    '# Log all kernel messages to the console.',
+    '# Logging much else clutters up the screen.',
+    '#kern.*                                                 /dev/console',
+    "kern.*                                                  $kernel_target",
```
Comment (ghoneycutt) [inline manifests/init.pp]: `${kernel_target}`

File: templates/rsyslog.conf.erb [hunk 5]
```diff
@@ -55,23 +55,9 @@ $ActionSendStreamDriverPermittedPeer <%= @permitted_peer %>
...
 
 # Everybody gets emergency messages
```
Comment (ghoneycutt) [inline templates/rsyslog.conf.erb]: What do you think about putting this logic in the manifest and then using a variable for the *.emerg target and handling that in the array, similar to how $kernel_target is handled?

Comment (Phil-Friderici) [inline templates/rsyslog.conf.erb]: Yeah, return the logic back into the manifest, I like it.
If you want that in the same commit I can add it.

Comment (ghoneycutt) [inline templates/rsyslog.conf.erb]: That would be awesome!

Surprised you are working on Fasching.

Comment (Phil-Friderici) [inline templates/rsyslog.conf.erb]: hahahaha...
Here we call it Karneval and as kid I loved it for the costumes and the action and never forget the free candies. But as grown up I can't stand the music anymore ;)

**Comment by Phil-Friderici** [comment]
Second commit enhances the spec tests depending on different rsyslog versions.

File: manifests/init.pp [hunk 6]
```diff
@@ -107,6 +115,15 @@
     }
   }
 
+  case $emerg_target {
```
Comment (ghoneycutt) [inline manifests/init.pp]: I might use if/else since there is only one match.. but whatever.

File: manifests/init.pp [hunk 7]
```diff
@@ -107,6 +115,15 @@
...
+    'USE_DEFAULTS': {
+      $emerg_target_real = $default_emerg_target
+    }
+    default: {
+      $emerg_target_real = $emerg_target
+    }
+  }
+
```
Comment (ghoneycutt) [inline manifests/init.pp]: should `validate_string($emerg_target_real)`

File: spec/classes/init_spec.rb [hunk 8]
```diff
@@ -232,52 +232,73 @@
...
+            :osfamily          => 'Debian',
+            :lsbmajdistrelease => '7',
           }
         end
+
         it do
           expect {
             should contain_class('rsyslog')
-          }.to raise_error(Puppet::Error,/^rsyslog_conf_version only knows <2>, <3>, <4>, <5>, <6>, <7>, <8> and <USE_DEFAULTS> as valid values and you have specified <invalid>/)
+          }.to raise_error(Puppet::Error)
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: place the error message here

File: manifests/init.pp (hunk 7 above)
Comment (ghoneycutt) [inline manifests/init.pp]: And the corresponding spec test..

**Comment by ghoneycutt** [comment]
Went ahead and made the updates - see PR #66

**Pull request closed without merging**